utilities/ride-model-util/src/main/java/com/adobe/ride/utilities/model/ModelObject.java

```diff
@@ -278,6 +278,43 @@ private void initializeModelObject(String resourceLocation, JSONObject presetNod
     loadModel(resourceLocation);
   }
 
+  /**
+   * Create a model object from a json schema string and use a predefined set of nodes to start
+   * with. Also only build a targeted set of nodes. The specified resource location is only used as
+   * metadata, so the content will not be fetched from it.
+   * 
+   * @param resourceLocation location of the schema within the project resources.
+   * @param modelString string which conforms to JSON schema standards
+   * @param presetNodes set of JSON nodes and values to be used in the object, not generated
+   *        dynamically
+   * @param nodesToBuild set of JSONnodes which are to be dynamically generated. Remaining schema
+   *        defined nodes will not be generated.
+   * @param useRequiredOnly boolean to determine whether to build only the required json nodes as
+   *        defined in the spec
+   */
+  public ModelObject(String resourceLocation, String modelString, JSONObject presetNodes,
+      Set<String> nodesToBuild, boolean useRequiredOnly) {
+    initializeModelObject(resourceLocation, modelString, presetNodes, nodesToBuild,
+        useRequiredOnly);
+  }
+
+  private void initializeModelObject(String resourceLocation, String modelString,
```

Comment:
I don't know that it makes sense to break this out into a sub method.  Might as well put it in the constructor.

Reply:
I did it for the other constructors in the class did the same. I'll inline it right away.